### PR TITLE
mafft 7.313 (new formula)

### PR DESCRIPTION
**Pull request by shohei1029**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/mafft.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    cause <<-EOS
+      Clang does not allow default arguments in out-of-line definitions of
+      class template members.
+      EOS
+  end
+
+  def install
+    make_args = %W[CC=#{ENV.cc} CXX=#{ENV.cxx} CFAGS=#{ENV.cflags}
+                   CXXFLAGS=#{ENV.cxxflags} PREFIX=#{prefix} MANDIR=#{man1}]
+    make_args << "ENABLE_MULTITHREAD=" if MacOS.version <= :snow_leopard
```
Comment (fxcoudert) [inline Formula/mafft.rb]: Let's remove the Snow Leopard case and caveat, and add `depends_on :macos => :lion`

File: Formula/mafft.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    make_args << "install"
```
Comment (fxcoudert) [inline Formula/mafft.rb]: Then `make_args` can be made a simple array, with no manipulation.

File: Formula/mafft.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <<-EOS
+        This build of MAFFT is not multithreaded on Snow Leopard
+        because its compiler does not support thread-local storage.
+      EOS
+    end
+  end
+
+  test do
+    (testpath/"test.fa").write ">1\nA\n>2\nA"
+    system "#{bin}/mafft", "test.fa"
```
Comment (fxcoudert) [inline Formula/mafft.rb]: Let's check the expected output please

File: Formula/mafft.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
+class Mafft < Formula
+  desc "Multiple alignments with fast Fourier transforms"
+  homepage "https://mafft.cbrc.jp/alignment/software/"
+  url "https://mafft.cbrc.jp/alignment/software/mafft-7.313-with-extensions-src.tgz"
+  sha256 "c48e5e05b427cae0d862daaef6148675d5ef57e24425c17b4c3d8da5b060eabd"
+
+  fails_with :clang do
+    build 421
+    cause <<-EOS
```
Comment (fxcoudert) [inline Formula/mafft.rb]: `~EOS` instead of `-EOS`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/mafft.rb (hunk 4 above)
Comment (fxcoudert) [reply Formula/mafft.rb]: Actually, let's remove the whole `fails_with` block, that clang version is _ancient_

**Comment by shohei1029** [comment]
Thanks a lot @fxcoudert. I reflected your useful comments.

**Pull request closed without merging**